NetKet/Hamiltonian/hamiltonian.hpp

```diff
@@ -41,6 +42,8 @@ class Hamiltonian : public AbstractHamiltonian {
         h_ = std::make_shared<Heisenberg<Graph>>(graph, pars);
       } else if (pars["Hamiltonian"]["Name"] == "BoseHubbard") {
         h_ = std::make_shared<BoseHubbard<Graph>>(graph, pars);
+      } else if (pars["Hamiltonian"]["Name"] == "Graph") {
+        h_ = std::make_shared<GraphHamiltonian<Graph>>(graph, pars);
```

Comment:
I've only now noticed it, but why do we store a `shared_ptr`?

Reply:
I think there was a problem with using 'unique_ptr' here, but I don't remember exactly what was the problem

Reply:
We are working on this in another PR, do not worry about this here